src/Stratis.Bitcoin.IntegrationTests/SegWitTests.cs

```diff
@@ -30,44 +28,66 @@ public void TestSegwitActivation()
                 coreNode.ConfigParameters.AddOrReplace("printtoconsole", "0");
                 coreNode.Start();
 
-                CoreNode node1 = builder.CreateStratisPowNode(true, fullNodeBuilder =>
+                CoreNode stratisNode = builder.CreateStratisPowNode(true, fullNodeBuilder =>
                 {
                     fullNodeBuilder
                         .UseConsensus()
                         .UseBlockStore()
+                        .UseWallet()
                         .UseMempool()
-                        .UseBlockNotification()
-                        .UseTransactionNotification()
                         .AddMining()
-                        .UseWallet()                        
-                        .UseApi()
                         .AddRPC();
                 });
 
-                WalletManager wm1 = node1.FullNode.NodeService<IWalletManager>() as WalletManager;
-                wm1.CreateWallet("Test1", "alice1");
-
-                RPCClient rpc1 = node1.CreateRPCClient();
+                RPCClient stratisNodeRpc = stratisNode.CreateRPCClient();
                 RPCClient coreRpc = coreNode.CreateRPCClient();
 
-                coreRpc.AddNode(node1.Endpoint, false);
-                rpc1.AddNode(coreNode.Endpoint, false);
+                coreRpc.AddNode(stratisNode.Endpoint, false);
+                stratisNodeRpc.AddNode(coreNode.Endpoint, false);
 
-                coreRpc.Generate(450);
+                // core does not mine segwit blocks by default so we disable the segwit auto activation on regtest
+                var prevSegwitDeployment = Network.RegTest.Consensus.BIP9Deployments[BIP9Deployments.Segwit];
+                Network.RegTest.Consensus.BIP9Deployments[BIP9Deployments.Segwit] = new BIP9DeploymentsParameters(1, 0, DateTime.Now.AddDays(50).ToUnixTimestamp());
 
-                BIP9DeploymentsArray bip9Constants = node1.FullNode.Network.Consensus.BIP9Deployments;
-                IConsensusLoop consensusLoop = node1.FullNode.NodeService<IConsensusLoop>();
-                ThresholdState[] bip9State = consensusLoop.NodeDeployments.BIP9.GetStates(node1.FullNode.Chain.Tip.Previous);
+                try
+                {
+                    // generate 450 blocks, block431 will have segwit activated.
+                    coreRpc.Generate(450);
 
-                Money amount = new Money(5.0m, MoneyUnit.BTC);
-                HdAddress destination1 = wm1.GetUnusedAddress(new WalletAccountReference("alice1", "account 0"));
+                    TestHelper.WaitLoop(() => stratisNode.CreateRPCClient().GetBestBlockHash() == coreNode.CreateRPCClient().GetBestBlockHash());
```

Comment:
I think this construct relies on consensus rather than the block store. It is conceivable that the entire test can complete before all the blocks have finished downloading. Maybe a separate test should test the application of the segwit consensus rules to the contents of the blocks.

Reply:
`GetBestBlockHash` will return the consensus best height if I am not mistaken.

Reply:
Ah, you are correct, disregard this.